nanachan/extensions/profiles.py

```diff
@@ -119,229 +104,328 @@ async def promo(self, ctx: Context):
+            required=False,
+            default=self.profile_dict['graduation_year'],
+        )
+        self.pronouns = ui.TextInput(
+            label='Pronouns',
+            placeholder='Enter your prnouns',
+            style=discord.TextStyle.short,
+            required=False,
+            default=self.profile_dict['pronouns'],
+        )
+        self.telephone = ui.TextInput(
+            label='Phone Number',
+            placeholder='Enter your phone number',
+            style=discord.TextStyle.short,
+            required=False,
+            default=self.profile_dict['telephone'],
+        )
+        self.add_item(self.birthday)
+        self.add_item(self.full_name)
+        self.add_item(self.graduation_year)
+        self.add_item(self.pronouns)
+        self.add_item(self.telephone)
+
+    async def on_submit(self, interaction: Interaction):
+        await interaction.response.defer()
+        errors = []
+        if (
+            self.birthday.value != ''
+            and re.fullmatch(r'^\d{4}-\d{2}-\d{2}$', self.birthday.value) is None
+        ):
+            errors.append('Invalid birthday format.')
+        if (
+            self.graduation_year.value != ''
+            and re.fullmatch(r'^\d{4}$', self.graduation_year.value) is None
+        ):
+            errors.append('Invalid gradutaion year.')
+        if (
+            self.telephone.value != ''
+            and re.fullmatch(r'^(\+33)|0\d{9}$', self.telephone.value) is None
+        ):
+            errors.append('Invalid phone number')
+        response = (
+            '\n'.join(errors) if len(errors) > 0 else 'All information gathered succesfully.'
+        )
+
+        def parse_date(date_str):
+            return datetime.strptime(date_str + ' +0000', '%Y-%m-%d %z') if date_str else None
+
+        def get_value_or_none(field):
+            return field.value or None
+
+        if not errors:
+            self.profile_dict.update(
+                {
+                    'birthday': parse_date(self.birthday.value),
+                    'full_name': get_value_or_none(self.full_name),
+                    'graduation_year': get_value_or_none(self.graduation_year),
+                    'pronouns': get_value_or_none(self.pronouns),
+                    'telephone': get_value_or_none(self.telephone),
+                }
+            )
+        await interaction.followup.send(response, ephemeral=True)
+
+
+class ProfileCreateOrChangeView(BaseView):
+    def __init__(self, bot: Bot, member: Member, profile: ProfileSearchResult):
+        super().__init__(bot)
+        self.member = member
+        self.profile = profile
+        n7_major_select = ui.Select(
+            placeholder='Select your major at N7',
+            options=[
+                SelectOption(emoji='⚡', label='3EA', value='3EA'),
+                SelectOption(emoji='🌊', label='MF2E', value='MF2E'),
+                SelectOption(emoji='💻', label='SN', value='SN'),
```

Comment:
only noticed now but that excludes older possible values (and we might have to change them in the future)